src/main/java/com/chartboost/sdk/sample/consent/AddConsentDialog.java

```diff
@@ -0,0 +1,175 @@
+            } else if (CCPA.CCPA_CONSENT.OPT_OUT_SALE.name().equals(consent)) {
+                dataUseConsent = new CCPA(CCPA.CCPA_CONSENT.OPT_OUT_SALE);
+            }
+        } else if (consentType == CONSENT_TYPE.COPPA) {
+            if ("true".equals(consent)) {
+                dataUseConsent = new COPPA(true);
+            } else if ("false".equals(consent)) {
+                dataUseConsent = new COPPA(false);
+            }
+        } else {
+            consent = editConsent.getText().toString();
+            dataUseConsent = new Custom(standard, consent);
+        }
+
+        if (dataUseConsent == null) {
+            Toast.makeText(getContext(), "Consent creation failed. Wrong standard or consent name.", Toast.LENGTH_SHORT).show();
+        } else {
+            if (!TextUtils.isEmpty(dataUseConsent.getPrivacyStandard()) && !TextUtils.isEmpty(dataUseConsent.getConsent().toString())) {
+                ConsentDialogListener listener = ((ConsentDialogListener) getActivity());
+                if(listener != null) {
+                    listener.createConsent(dataUseConsent);
+                } else {
+                    Toast.makeText(getContext(), "Consent creation failed.", Toast.LENGTH_SHORT).show();
+                }
+            } else {
+                Toast.makeText(getContext(), "Consent creation failed. Wrong standard or consent name.", Toast.LENGTH_SHORT).show();
+            }
+        }
+
+        dismiss();
+    }
+
+    private void switchView() {
+        if (consentType == CONSENT_TYPE.GDPR) {
+            editStandard.setVisibility(View.GONE);
+            editConsent.setVisibility(View.GONE);
+            autocompleteConsent.setVisibility(View.VISIBLE);
+            setAutoCompleteAdapter(gdprConsents);
+        } else if (consentType == CONSENT_TYPE.CCPA) {
+            editStandard.setVisibility(View.GONE);
+            editConsent.setVisibility(View.GONE);
+            autocompleteConsent.setVisibility(View.VISIBLE);
+            setAutoCompleteAdapter(ccpaConsents);
+        } else if (consentType == CONSENT_TYPE.COPPA) {
+            editStandard.setVisibility(View.GONE);
+            editConsent.setVisibility(View.GONE);
+            autocompleteConsent.setVisibility(View.VISIBLE);
+            setAutoCompleteAdapter(coppaConsents);
+        } else {
+            editStandard.setVisibility(View.VISIBLE);
+            editConsent.setVisibility(View.VISIBLE);
+            autocompleteConsent.setVisibility(View.GONE);
+        }
+    }
+
+    private void setAutoCompleteAdapter(List<String> data) {
+        ArrayAdapter<String> adapter = new ArrayAdapter<>(getContext(), android.R.layout.simple_list_item_1, data);
+        autocompleteConsent.setAdapter(adapter);
+        autocompleteConsent.showDropDown();
+    }
+
+
+    public static AddConsentDialog newInstance() {
+        return new AddConsentDialog();
+    }
+
+    public enum CONSENT_TYPE {
```

Comment:
Wrong naming for an enum per Java convention

Reply:
yes i was moving the code from kotlin back to java for this sample app :P